sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala

```diff
@@ -519,6 +519,8 @@ object FunctionRegistry {
     expressionBuilder("mode", ModeBuilder),
     expression[HllSketchAgg]("hll_sketch_agg"),
     expression[HllUnionAgg]("hll_union_agg"),
+    expression[Product]("product"),
```

Comment:
The changes in this file actually increase our surface area. So I'd like to have some feedback and understand if this is ok.

Reply:
I remember there was some reason to not add `product` here @cloud-fan 

https://github.com/apache/spark/pull/38915#discussion_r1039400753

Reply:
I don't have a strong opinion now, as we are going for feature parity between SQL and DataFrame APIs recently. cc @HyukjinKwon

Reply:
Actually yeah. Those methods were intentionally not added to SQL expressions.

Reply:
Ok, let me try something different.

Reply:
I will create a separate registry for these functions, and only use them in Dataframe resolution.

Reply:
Tracked in https://issues.apache.org/jira/browse/SPARK-49004.